### ISSUE TITLE
Error using a function as part of a macro

### ISSUE THREAD

**Issue by gpanders** [bakpakin/Fennel#383]
Here is a minimal example:

```fennel
(macro test [func]
  (let [form (list)]
    (table.insert form `(tset _G "test" ,func))
    form))

(test (fn [] (print "Hello world")))
```

Compiling this throws the following error:

```
Compile error in unknown:?
  cannot call literal value (tset _G test (fn table: 0x12291f6f0 (print Hello world)))

* Try checking for typos.
* Try checking for a missing function name.
```

The argument list for the function is being wrongly converted to its string representation.

If I change the macro to use

```fennel
(let [form `(do)]
```

then it works. I'm not sure if this is a bug or not; perhaps this is intended behavior and I'm just misunderstanding the macro system.

**Comment by technomancy** [comment]
This is one case where it's not clear what's going on until you look at
the macroexpansion.

```fennel
(macro test [func]
  (let [form (list)]
    (table.insert form `(tset _G "test" ,func))
    form))

(macrodebug (test (fn [] (print "Hello world"))))

;; -> ((tset _G "test" (fn {} (print "Hello world"))))
```

Does that make sense now?

> The argument list for the function is being wrongly converted to its string representation.

However, this is legitimately a bug in the error reporting, but it's
unrelated to the failure to compile.

**Comment by gpanders** [comment]
Thanks for that explanation. Yes, that makes sense, and I had forgotten about `macrodebug`.

>>The argument list for the function is being wrongly converted to its string representation.
>
>However, this is legitimately a bug in the error reporting, but it's
>unrelated to the failure to compile.

Should I keep the issue open for this?

**Comment by technomancy** [comment]
I just pushed a fix in c655ccd so I think we can close it out now.
